contracts/deposit-contracts/contract/src/main.rs

```diff
@@ -0,0 +1,214 @@
+    let temp_purse: URef = runtime::get_named_arg(RUNTIME_ARG_TEMP_PURSE);
+    let amount: U512 = runtime::get_named_arg(RUNTIME_ARG_AMOUNT);
+    let deposit_purse_uref: URef = runtime::get_key(KAIROS_DEPOSIT_PURSE)
+        .unwrap_or_revert_with(ApiError::MissingKey)
+        .into_uref()
+        .unwrap_or_revert_with(ApiError::UnexpectedKeyVariant);
+    system::transfer_from_purse_to_purse(temp_purse, deposit_purse_uref, amount, None)
+        .unwrap_or_revert();
+
+    let most_recent_deposit_counter_uref = runtime::get_key(KAIROS_MOST_RECENT_DEPOSIT_COUNTER)
+        .unwrap_or_revert_with(ApiError::MissingKey)
+        .into_uref()
+        .unwrap_or_revert();
+    let mut most_recent_deposit_counter_value: u64 =
+        storage::read(most_recent_deposit_counter_uref)
+            .unwrap_or_revert()
+            .unwrap_or_revert();
+    let new_deposit_record: Deposit = Deposit {
+        account: get_immediate_caller().unwrap_or_revert(),
+        amount,
+        timestamp: None,
+        processed: false,
+    };
+
+    let kairos_deposit_event_dict_uref = runtime::get_key(KAIROS_DEPOSIT_EVENT_DICT)
+        .unwrap_or_revert_with(ApiError::MissingKey)
+        .into_uref()
+        .unwrap_or_revert();
+    storage::dictionary_put::<Vec<u8>>(
+        kairos_deposit_event_dict_uref,
+        &most_recent_deposit_counter_value.to_string(),
+        bincode::serialize(&new_deposit_record).unwrap(),
+    );
+
+    most_recent_deposit_counter_value += 1u64;
+    storage::write(
+        most_recent_deposit_counter_uref,
+        most_recent_deposit_counter_value,
+    );
+}
+
+#[no_mangle]
+pub extern "C" fn incr_last_processed_deposit_counter() {
+    sec_check(vec![SecurityBadge::Admin]);
+    let last_processed_deposit_counter_uref =
+        runtime::get_key(KAIROS_LAST_PROCESSED_DEPOSIT_COUNTER)
+            .unwrap_or_revert_with(ApiError::MissingKey)
+            .into_uref()
+            .unwrap_or_revert();
+    let mut last_processed_deposit_counter_value: u64 =
+        storage::read(last_processed_deposit_counter_uref)
+            .unwrap_or_revert()
+            .unwrap_or_revert();
+    last_processed_deposit_counter_value += 1u64;
+    storage::write(
+        last_processed_deposit_counter_uref,
+        last_processed_deposit_counter_value,
+    );
+}
+
+#[no_mangle]
+pub extern "C" fn withdrawal() {
+    sec_check(vec![SecurityBadge::Admin]);
+    let destination_purse: URef = runtime::get_named_arg(RUNTIME_ARG_DEST_PURSE);
+    let amount: U512 = runtime::get_named_arg(RUNTIME_ARG_AMOUNT);
+    let deposit_purse: URef = runtime::get_key(KAIROS_DEPOSIT_PURSE)
+        .unwrap_or_revert_with(ApiError::MissingKey)
+        .into_uref()
+        .unwrap_or_revert()
+        .with_access_rights(AccessRights::READ_ADD_WRITE);
+    system::transfer_from_purse_to_purse(deposit_purse, destination_purse, amount, None)
+        .unwrap_or_revert();
+}
+
+#[no_mangle]
+pub extern "C" fn change_security() {
```

Comment:
I would like to see documentation on when this will/should be called and what it does.